src/QMCWaveFunctions/TWFPrototype.cpp

```diff
@@ -0,0 +1,316 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2021 QMCPACK developers.
+//
+// File developed by:   Raymond Clay III, rclay@sandia.gov, Sandia National Laboratories
+//
+// File created by:   Raymond Clay III, rclay@sandia.gov, Sandia National Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "QMCWaveFunctions/TWFPrototype.h"
+#include "Numerics/DeterminantOperators.h"
+#include <iostream>
+namespace qmcplusplus
+{
+TWFPrototype::TWFPrototype() : initialized(false) { std::cout << "TWFPrototype Constructed\n"; }
+
+TWFPrototype::IndexType TWFPrototype::get_group_index(const IndexType gid)
+{
+  IndexType ndets = groups.size();
```

Comment:
what is the source of truth as far an ndets go?  This code is all over the place on that.

Reply:
I did a lot of cleaning and renaming here.  "Determinants" is sloppy, and has been renamed to "groups".  Groups are collections of particles which for this TWF object are antisymmetric under particle exchange, hence why the naming got sloppy.  Hopefully this is fixed and is more consistent now.